editor/plugins/animation_player_editor_plugin.cpp

```diff
@@ -824,66 +739,81 @@ void AnimationPlayerEditor::_update_animation() {
 
 void AnimationPlayerEditor::_update_player() {
 	updating = true;
-	List<StringName> animlist;
-	if (player) {
-		player->get_animation_list(&animlist);
-	}
 
 	animation->clear();
 
-#define ITEM_DISABLED(m_item, m_disabled) tool_anim->get_popup()->set_item_disabled(tool_anim->get_popup()->get_item_index(m_item), m_disabled)
-
-	ITEM_DISABLED(TOOL_SAVE_ANIM, animlist.size() == 0);
-	ITEM_DISABLED(TOOL_SAVE_AS_ANIM, animlist.size() == 0);
-	ITEM_DISABLED(TOOL_DUPLICATE_ANIM, animlist.size() == 0);
-	ITEM_DISABLED(TOOL_RENAME_ANIM, animlist.size() == 0);
-	ITEM_DISABLED(TOOL_EDIT_TRANSITIONS, animlist.size() == 0);
-	ITEM_DISABLED(TOOL_COPY_ANIM, animlist.size() == 0);
-	ITEM_DISABLED(TOOL_REMOVE_ANIM, animlist.size() == 0);
-
-	stop->set_disabled(animlist.size() == 0);
-	play->set_disabled(animlist.size() == 0);
-	play_bw->set_disabled(animlist.size() == 0);
-	play_bw_from->set_disabled(animlist.size() == 0);
-	play_from->set_disabled(animlist.size() == 0);
-	frame->set_editable(animlist.size() != 0);
-	animation->set_disabled(animlist.size() == 0);
-	autoplay->set_disabled(animlist.size() == 0);
-	tool_anim->set_disabled(player == nullptr);
-	onion_toggle->set_disabled(animlist.size() == 0);
-	onion_skinning->set_disabled(animlist.size() == 0);
-	pin->set_disabled(player == nullptr);
-
 	if (!player) {
 		AnimationPlayerEditor::get_singleton()->get_track_editor()->update_keying();
 		return;
 	}
 
+	List<StringName> libraries;
+	if (player) {
+		player->get_animation_library_list(&libraries);
+	}
+
 	int active_idx = -1;
-	for (const StringName &E : animlist) {
-		animation->add_item(E);
+	bool no_anims_found = true;
 
-		if (player->get_assigned_animation() == E) {
-			active_idx = animation->get_item_count() - 1;
+	for (const StringName &K : libraries) {
+		if (K != StringName()) {
+			animation->add_separator(K);
+		}
+
+		Ref<AnimationLibrary> library = player->get_animation_library(K);
+		List<StringName> animlist;
+		library->get_animation_list(&animlist);
+
+		for (const StringName &E : animlist) {
+			String path = K;
+			if (path != "") {
+				path += "/";
+			}
+			path += E;
+			animation->add_item(path);
+			if (player->get_assigned_animation() == path) {
+				active_idx = animation->get_selectable_item(true);
+			}
+			no_anims_found = false;
 		}
 	}
+#define ITEM_DISABLED(m_item, m_disabled) tool_anim->get_popup()->set_item_disabled(tool_anim->get_popup()->get_item_index(m_item), m_disabled)
+
+	ITEM_DISABLED(TOOL_DUPLICATE_ANIM, no_anims_found);
+	ITEM_DISABLED(TOOL_RENAME_ANIM, no_anims_found);
+	ITEM_DISABLED(TOOL_EDIT_TRANSITIONS, no_anims_found);
+	ITEM_DISABLED(TOOL_REMOVE_ANIM, no_anims_found);
```

Comment:
Remember to `#undef` macros after use.
```suggestion

#define ITEM_DISABLED(m_item, m_disabled) tool_anim->get_popup()->set_item_disabled(tool_anim->get_popup()->get_item_index(m_item), m_disabled)
	ITEM_DISABLED(TOOL_DUPLICATE_ANIM, no_anims_found);
	ITEM_DISABLED(TOOL_RENAME_ANIM, no_anims_found);
	ITEM_DISABLED(TOOL_EDIT_TRANSITIONS, no_anims_found);
	ITEM_DISABLED(TOOL_REMOVE_ANIM, no_anims_found);
#undef ITEM_DISABLED
```

Though it's arguable how useful this simple macro is since it's only used 4 times. It might be more relevant to actually cache the value of `tool_anim->get_popup()` and do:
```suggestion
	{
		Popup *p = tool_anim->get_popup();
		p->set_item_disabled(p->get_item_index(TOOL_DUPLICATE_ANIM), no_anims_found));
		p->set_item_disabled(p->get_item_index(TOOL_RENAME_ANIM), no_anims_found));
		p->set_item_disabled(p->get_item_index(TOOL_EDIT_TRANSITIONS), no_anims_found));
		p->set_item_disabled(p->get_item_index(TOOL_REMOVE_ANIM), no_anims_found));
	}
```

Reply:
This was like this before for the most part, so I did not bother to change it. There are less menu options now so I guess it makes less sense to have it, but again no strong feeling either way.